### ISSUE TITLE
enhance module xml descriptions, same for subsystems?

### ISSUE THREAD

**Issue by flixr** [paparazzi/paparazzi#150]
Add a `<doc>` node with also `<define>` and `<configure>`, and the possible values for each to the module xml file.
Then you have the description, and can do a sanity check on the configuration options and easily list the available ones.

We should then probably also include the appropriate settings files, as per issue #23

The question then is if we really want to keep both, subsystems and modules... as it is only a semantic difference with modules representing optional part and subsystems mandatory parts. (ok, and they have different directories, etc.)
Basically if you write a module that doesn't use the module periodic and event functions, and instead it's functions get directly called from main, you have a subsystem...
